apps/api/src/controllers/v1/types.ts

```diff
@@ -467,15 +471,31 @@ export const extractV1Options = z
       .refine(
         (val) => {
           if (!val) return true; // Allow undefined schema
+          
+          // Check if this is an unconverted Zod schema
+          if (val && typeof val === 'object' && val._def) {
+            console.error('Schema validation error: Detected unconverted Zod schema with _def property:', {
+              type: val._def?.typeName,
+              keys: Object.keys(val),
+              hasZodProperties: !!(val._def || val.parse || val.safeParse)
+            });
```

Comment:
`console.error` instead of `logger.error` is not allowed, logging should not be done here as it will lead to log flooding on production.